duckduckgo/iOS
# NetP waitlist final touches

Task/Issue URL: https://app.asana.com/0/0/1206027185389966/f
Tech Design URL:
CC:

**Description**:

This PR makes some final touches for the NetP waitlist, including:

* Adding RMF support
* Adding pixels
* Updates NetP to only be visible to US users (unless they're internal)
* Recording the date at which the user activated NetP, in order to show messages remotely if necessary



**Steps to test this PR**:

This PR is using a custom RMF URL which matches NetP users in the US who have used the feature for at least 3 days. To test this properly may require overriding some of these values.

To override the matching values, see the `fetchAndProcess` function inside `RemoteMessaging.swift`. The `isNetworkProtectionWaitlistUser` and `daysSinceNetworkProtectionEnabled` variables should be changed to test various conditions.

Note that RMF only processes messages once a day - if you change the values above, this will not necessarily change what you see. To override this, go to `RemoteMessagingConfigProcessor.swift` and change `shouldProcessConfig` to always return true.

1. After following the instructions above, you should begin to get a remote message visible in the app; check that you do, and that the URL is opened with survey parameters included
2. Run through the waitlist flow and check that pixels get fired along the way - note that these are daily pixels, we unfortunately have no way to reset these right now except for uninstalling and reinstalling the app to clear user defaults



**Copy Testing**:


**Orientation Testing**:


**Device Testing**:


**OS Testing**:


**Theme Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
